### PR TITLE
Fixes buttbots not hearing/using local speech, fixes spammable buttbot

### LINKED ISSUE #11865: Buttbot's only seem to hear through intercoms and radios
Revision: 6db9bf47a602d82e47355c0af6eea24446cd1ab2 

Talking to a buttbot doesn't make it repeat what you're saying
UNLESS it hears what you're saying through the radio/intercom.

### LINKED ISSUE #17221: Issues with Buttbots
#### Description of issue
Buttbots no longer do their usual behavior of parroting and replacing the text with "butt". Instead, they only say "butt" and fart (like when they're clicked).

Buttbots also can be spam clicked on to flood the chat (not radio). As funny as this is, it's unintentional and is due to a lack of sincelastfart check on attack_hand.

#### Difference between expected and actual behavior

Expected behavior: Buttbots should parrot what the user says and repeat it with "butt" replaced. Then fart. They should also have a cooldown on clicks.

Actual behavior: Buttbots don't parrot and instead just say "butt" and fart when they hear something. They do not have a cooldown on clicks.

#### Steps to reproduce 
Talk to a Buttbot. Spam click on a Buttbot.

#### Specific information for locating

code/game/machinery/bots/buttbot.dm

/obj/machinery/bot/buttbot

#### Length of time in which bug has been known to occur

I don't know. Seems to be since #17116.

#### Server revision

Only tested on local, not an actual server.

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [ ] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

I need someone else to test this in a current round while I try and fix this, thanks. butt

### PR DESCRIPTION
**Pull request by gbasood**
Fixes #17221 
Fixes #11865

### CODE REVIEW

**Comment by Mothblocks** [comment]
Long live Buttbots. I'll work on #17054 when this is merged.

File: code/game/machinery/bots/buttbot.dm [hunk 1]
```diff
@@ -26,7 +26,14 @@ Here it is: Buttbot.
...
 		return
-	fart()
+	if(can_fart())
+		speak("butt")
+		fart()
+
+/obj/machinery/bot/buttbot/proc/can_fart()
+	if(sincelastfart + 5 < world.timeofday)
+		return TRUE
+	return FALSE
```
Comment (Mothblocks) [inline code/game/machinery/bots/buttbot.dm]: Why not just `return sincelastfart + 5 < world.timeofday`?

Comment (gbasood) [reply code/game/machinery/bots/buttbot.dm]: o ya

**Comment by clusterfack** [comment]
Ohhh it was the sleep that makes sense

File: code/game/machinery/bots/buttbot.dm [hunk 2]
```diff
@@ -36,16 +41,17 @@ Here it is: Buttbot.
...
 /obj/machinery/bot/buttbot/proc/fart()
-	if(sincelastfart + 5 < world.timeofday)
-		speak("butt")
+	if(can_fart())
 		playsound(get_turf(src), 'sound/misc/fart.ogg', 50, 1)
 		sincelastfart = world.timeofday
 
 /obj/machinery/bot/buttbot/Hear(var/datum/speech/speech, var/rendered_speech="")
 	set waitfor = 0 //Buttbots speaking should be queued after the original speech completes
-	if(prob(buttchance) && !findtext(speech.message,"butt"))
```
Comment (Jared-Fogle) [inline code/game/machinery/bots/buttbot.dm]: sneaky c*ders

**Pull request merged**